### ISSUE TITLE
commands.json not getting refreshed properly.

### ISSUE THREAD

**Issue by veleek** [madskristensen/CommandTaskRunner#4]
When I update `commands.json` to add an additional parameter to the script (`-NoProfile` in this case), the command doesn't seem to get refreshed.  I need to rename the command in order for it to get picked up as a new command, otherwise the original command gets run.

1. Create a command, e.g.
    "TestCommand": {
      "fileName": "powershell.exe",
      "workingDirectory": ".",
      "arguments": "-Command & {  Write-Host 'Test 1' }"
    }
2. Save `commands.json`
  - If you already have a list of commands, TestCommand automatically appears in the list. 
3. Right click command -> Run Command
  - **Output:** Test 1
4. Update command output to 'Test 2' and save `commands.json`.
5. Run again.
  - **Output:** Test 1
6. Rename command to "TestCommand2" and save `commands.json`.
  - TestCommand2 appears in the Task Runner Explorer
7. Run TestCommand2
  - **Output:** Test 2

**Comment by spayton** [comment]
I get the same behaviour, I have to restart VS to get it to use the command, or duplicate it to new command as veleek says above.

**Comment by nicollaas94** [comment]
If u rename the command name it will be refreshed ...

**Comment by solvingj** [comment]
@nicollaas94 this does not work. 

@madskristensen There are two bugs related to string value caching. 

1.  Renamed commands show up with new names in the TRE window, but if you change the command text or args text, visual studio will not use the new strings.  It is improperly caching them and not looking at the `commands.json` file for those changes specifically.  Indeed, you have to close/reopen solution. 

2. All of the special MSVC variables which get "string replaced" are only evaluated when the first command is invoked, and never checked after.  So, for example with the variable `$(ConfigurationName)`... If it is accessed while Visual Studio GUI has `Debug` configuration selected, the string will be replaced with `Debug` properly.  However, if you then change the configuration to `Release`, it the `$(ConfigurationName)` string will not be re-evaluated and re-replaced in future executions of commands.  It will always say `Debug` unless you reload the solution.

https://github.com/madskristensen/CommandTaskRunner/blob/de2b303e603c341dad3a5133e2b877352a8d3b32/src/TaskRunner/TaskRunnerProvider.cs#L111

Basically, `SetVariables` needs to be invoked on every execution of a command, and I don't think it is.

**Comment by madskristensen** [comment]
There are limitations in the TRX itself that makes this hard to work around for any extension
